### PR TITLE
feat: add useButtonStyles

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR adds a convenience hook abstracting the recurring pattern of using our button theme styles on something that's not a button. By using theme styles, we avoid having to find the right `useColorMode()` values and possibly getting them wrong.

Example consumption in the stacked PR https://github.com/shapeshift/web/pull/3134/commits

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

This might be useless and while semantically weird, we might get away with `<Button as ={SomeOtherComponent} />`

### Engineering

- Implementation is sane

### Operations

- None, this isn't used yet

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3133** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3133" target="_blank">Graphite</a> 
    * **PR #3135** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3135" target="_blank">Graphite</a>  👈
      * **PR #3134** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3134" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3135?utm_source=stack-comment).

File: src/hooks/useButtonStyles/useButtonStyles.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export const useButtonStyles = ({ variant, colorScheme }: UseButtonStyleProps) => {
+  const {
+    components: {
+      Button: { variants },
+    },
+  } = useTheme()
+
+  const { [variant]: buttonVariant } = variants as Record<string, SystemStyleFunction>
```
Comment (gomesalexandre) [inline src/hooks/useButtonStyles/useButtonStyles.ts:18]: `useTheme()` isn't really typed, there's a generic for it but all it does is taking a full dictionary object to be used as a return type
https://github.com/chakra-ui/chakra-ui/blob/3e1b3f6b6a7398b71ac08339110f075695fbae94/packages/components/system/src/use-theme.ts#L11

File: src/hooks/useButtonStyles/useButtonStyles.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  const { colorMode } = useColorMode()
+
+  const styleProps = buttonVariant({
+    colorScheme,
+    theme,
+    colorMode,
```
Comment (gomesalexandre) [inline src/hooks/useButtonStyles/useButtonStyles.ts:25]: The guts of this convenience hook, currying the `buttonVariant` call with these two

File: src/hooks/useButtonStyles/useButtonStyles.ts (hunk 1 above)
Comment (0xApotheosis) [reply src/hooks/useButtonStyles/useButtonStyles.ts:18]: I was just about to ask re: this, ty

**Review by 0xApotheosis — APPROVED** [review]
Tidy.

 🍯

**Pull request merged**